akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -166,7 +174,10 @@ private[akka] class TLSActor(settings: ActorMaterializerSettings,
       case Some(TLSClientAuth.Need) ⇒ engine.setNeedClientAuth(true)
       case _                        ⇒ // do nothing
     }
-    params.sslParameters foreach (p ⇒ engine.setSSLParameters(p))
+    params.sslParameters foreach { p ⇒
+      hostInfo foreach { case (host, _) ⇒ applySNI(host, engine, p) }
```

Comment:
(Needs to be applied in `applySessionParameters`, it's not enough to do it just when we obtain the `engine`)